lib/posix/clock.c

```diff
@@ -34,6 +51,12 @@ int clock_gettime(clockid_t clock_id, struct timespec *ts)
 		break;
 
 	case CLOCK_REALTIME:
+#ifdef CONFIG_RTC
+		dev_rtc = clock_getrtc();
+		if (dev_rtc) {
```

Comment:
so if something goes wrong with rtc device we silently jump to use uptime. That should somehow be reported.

Reply:
I rewrote this